packages/ns-plug/files/ns-plug

```diff
@@ -112,6 +112,7 @@ dev tun-nsplug
 tls-client
 script-security 2
 route-up /usr/sbin/ns-controller-push-info
+down /usr/sbin/restart-ns-plug
```

Comment:
Why not calling the init.d directly?
Reading the man, I'm now sure which is the right syntax.
This:
```
down /etc/init.d/ns-plug restart
```

or this:
```
down "/etc/init.d/ns-plug restart"
```

Reply:
I tried and it does not work, `down /etc/init.d/ns-plug restart`, I got a warning, the second is not executed

I could test again if you want

My best question is, do we try to increase the time to test the remote server, 60 seconds here is enough in home lab, maybe a bigger value could fit better

Reply:
> I could test again if you want

Yes please, from the man it seems it should work:
> --down cmd
Run command cmd after TUN/TAP device close (post --user UID change and/or --chroot ). cmd consists of a path to script (or executable program), optionally followed by arguments. The path and arguments may be single- or double-quoted and/or escaped using a backslash, and should be separated by one or more spaces.Called with the same parameters and environmental variables as the --up option above.
Note that if you reduce privileges by using --user and/or --group, your --down script will also run at reduced privilege.

> My best question is, do we try to increase the time to test the remote server, 60 seconds here is enough in home lab, maybe a bigger value could fit better

The timeout is for the first connection, then with your patch the ns-plug script is restarted and a new interval of 60 seconds is tried again.

Please also verify that the new patch does not prevent to shutdown the service using /etc/init.d/ns-plug stop.

Reply:
we need I think a script, mainly we need to wait the service is fully stopped, it takes a while, at least 5 seconds on my VM

Reply:
I'v done some tests, this works:
```
down "/etc/init.d/ns-plug restart"
```

But it has a drawback: the service can't be stopped using `/etc/init.d/ns-plug stop`